src/Jackett.Common/Indexers/HDBitsApi.cs

```diff
@@ -0,0 +1,203 @@
+            AddCategoryMapping(1, TorznabCatType.Movies, "Movie");
+            AddCategoryMapping(4, TorznabCatType.Audio, "Music");
+            AddCategoryMapping(5, TorznabCatType.TVSport, "Sport");
+            AddCategoryMapping(2, TorznabCatType.TV, "TV");
+            AddCategoryMapping(7, TorznabCatType.XXX, "XXX");
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            dynamic requestData = new JObject();
+
+            try
+            {
+                await MakeApiRequest("test", requestData);
+            }
+            catch (Exception e)
+            {
+                throw new ExceptionWithConfigData(e.Message, configData);
+            }
+
+            IsConfigured = true;
+
+            SaveConfig();
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            dynamic requestData = new JObject();
+            var queryString = query.GetQueryString();
+
+            if (!string.IsNullOrWhiteSpace(query.ImdbID))
+            {
+                requestData["imdb"] = new JObject();
+                requestData["imdb"]["id"] = query.ImdbID;
+            }
+
+            if (!string.IsNullOrWhiteSpace(queryString))
+            {
+                requestData["search"] = queryString;
+            }
+
+            var categories = MapTorznabCapsToTrackers(query);
+
+            if (categories.Count > 0)
+            {
+                requestData["category"] = new JArray();
+
+                foreach (var cat in categories)
+                {
+                    requestData["category"].Add(new JValue(cat));
+                }
+            }
+
+            requestData["limit"] = 100;
+
+            var response = await MakeApiRequest("torrents", requestData);
+            var releases = new List<ReleaseInfo>();
+
+            foreach (JObject r in response["data"])
+            {
+                var release = new ReleaseInfo();
+                release.Title = (string)r["name"];
+                release.Comments = new Uri(SiteLink+"details.php?id="+(string)r["id"]);
+                release.Link = new Uri(SiteLink+"download.php/"+(string)r["filename"]+"?id="+(string)r["id"]+"&passkey="+configData.Passkey.Value);
+                release.Guid = release.Link;
+
+                if (r.ContainsKey("imdb"))
+                {
+                    release.Imdb = ParseUtil.GetImdbID((string)r["imdb"]["id"]);
+                }
+
+                if (r.ContainsKey("tvdb"))
+                {
+                    release.TVDBId = (long)r["tvdb"]["id"];
+                }
+
+                release.UploadVolumeFactor = 1;
+                int[] mediumsFor50 = {1,5,4};
+
+                // 100% Neutral Leech: all XXX content.
+                if ((int)r["type_category"] == 7)
```

Comment:
the rules to apply freeleech can change...
where did you find that, in JS code? is the freeleech informed in the json response?

Reply:
There's a `freeleech` property in the response, but it's only `true` if the torrent is 100% free leech. The other kinds of freeleech (25%, 50%), as well as what HDBits call "100% neutral leech", is described in their rules - that's what I've based the implementation on.

Reply:
There's a `freeleech` property in the response, but it's only `true` if the torrent is 100% free leech. The other types of freeleech (25% and 50%), as well as what HDBits call "100% neutral leech", is described in their rules - that's what I've based the implementation on.

Reply:
ok